sdk/core/core-http/lib/operationOptions.ts

```diff
@@ -20,6 +20,22 @@ export interface OperationOptions {
   tracingOptions?: OperationTracingOptions;
 }
 
+const OperationOptionsValues: Set<string> = new Set<keyof OperationOptions>([
+  "abortSignal",
+  "requestOptions",
+  "tracingOptions"
+]);
+
+/**
+ * Returns true if `options` adheres to the `OperationOptions` interface.
+ * @param options  An options bag that is either an `OperationOptions` or a `RequestOptionsBase`
+ */
+export function isOperationOptions(
+  options: OperationOptions | RequestOptionsBase
+): options is OperationOptions {
+  return Object.keys(options).every((value) => OperationOptionsValues.has(value));
```

Comment:
>Object.keys(options).every((value) => OperationOptionsValues.has(value)); [](start = 9, length = 73)

So this would not allow types that extend `OperationOptions` and add new properties?

Reply:
You are right! 

Hmm, so if I understand correctly, the way this works today is that the configuration in `OperationSpec` tells sendOperationRequest how to retrieve parameter values, so we definitely want to allow this to be extended.

Maybe @bterlson was right and we need a new method that takes `OperationOptions` and then we can fix the codegen to call that instead?